### PR TITLE
Feature/cedux init returns store

### PR DESCRIPTION
**Pull request by JSchaenzle**

### CODE REVIEW

File: cedux.h [hunk 1]
```diff
@@ -14,15 +14,15 @@ LIST_DECLARATION(STORE_NAME##_reducer_list, STORE_NAME##_REDUCER, 32)
...
-void cedux_init_##STORE_NAME(struct STORE_NAME##_handle * p_store);                             \
+struct STORE_NAME##_handle cedux_init_##STORE_NAME(void);                                       \
 void cedux_dispatch_##STORE_NAME(struct STORE_NAME##_handle * p_store, ACTION_TYPE action);     \
 bool cedux_run_##STORE_NAME(struct STORE_NAME##_handle * p_store);                              \
 struct STORE_NAME##_handle                                                                      \
 {                                                                                               \
   TREE_TYPE tree;                                                                               \
   struct STORE_NAME##_action_queue action_queue;                                                \
   struct STORE_NAME##_reducer_list reducer_list;                                                \
-} STORE_NAME;
```
Comment (JSchaenzle) [inline cedux.h:25]: This line is the key change. The declare store macro no longer declares an instance of the store.

**Comment by JSchaenzle** [comment]
Creating this PR to get some visibility to this change. I think it makes the library cleaner since users are now in control of declaring the actual store where ever they want to.

**Comment by jmgregory** [comment]
Very cool library!  This change makes sense to me.

With this PR, would the new store name have to match anything in the macro declaration?  In other words, could I do:

```
struct my_store_handle my_store = cedux_init_my_store();
struct my_store_handle my_store2 = cedux_init_my_store();
struct my_store_handle something_different = cedux_init_my_store();
```

**Comment by JSchaenzle** [comment]
@jmgregory Thanks!  Yes, your example would work. However, those are all the same type of store because the name `my_store` is the same for all. If you wanted different stores that contain different types of data structures you could do that like this:
```c
CEDUX_DECLARE_STORE(struct account_state_tree, struct account_state_actions, account_store);
CEDUX_DEFINE_STORE(struct account_state_tree, struct account_state_actions, account_store);

CEDUX_DECLARE_STORE(struct app_state_tree, struct app_state_actions, app_state_store);
CEDUX_DEFINE_STORE(struct app_state_tree, struct app_state_actions, app_state_store);

struct account_state_store_handle account_store = cedux_init_account_store();
struct app_state_store_handle = cedux_init_app_state_store();  
```
Notice the init method names change.

**Pull request merged**